DISCO methods paper, initial attempt to address feedback from reviewers 2&3 (#1033)
greenelab/covid19-review: content/60.methods.md

Similarly, many literature reviews have been written on the available COVID-19 literature [@doi:10.1016/j.molmed.2020.02.008; @doi:10.1016/j.immuni.2020.05.002; @doi:10.1126/scitranslmed.abc1931; @doi:10.1001/jama.2020.6019; @doi:10.1038/d41591-020-00026-w; @doi:10.1001/jama.2020.12839], but static reviews quickly become outdated as new research is released or existing research is retracted or superseded.
One example is a review of topics in COVID-19 research including vaccine development [@doi:10.1001/jama.2020.12839].
This review was published on July 10, 2020, four days before Moderna released the surprisingly promising results of their phase 1 trial [@doi:10.1056/NEJMoa2022483] that changed expectations surrounding vaccines.
Therefore, the COVID-19 publishing climate presented a challenge where curation of the literature by a diverse group of experts in a format that could respond quickly to high-volume, high-velocity information was desirable.

We therefore sought to develop a platform for scientific discussion and collaboration around COVID-19 by adapting open publishing infrastructure to accommodate the scale of the COVID-19 publishing boom.

> Recent advances in open publishing have created an infrastructure that facilitates distributed, version-controlled collaboration on manuscripts [@doi:10.1371/journal.pcbi.1007128].<!--To Do: cite other efforts here & explain why Manubot was the best fit-->

Reviewer: Some initial text you can work from.  Maybe this would fit as a new paragraph following this one.

> We selected Manubot because it offers several advantages over comparable collaborative writing platforms such as Authorea, Overleaf, Google Docs, Word Online, or wikis [@doi:10.1371/journal.pcbi.1007128].
Citation-by-identifier ensures consistent reference metadata standards that would be difficult to maintain manually in a manuscript with dozens of authors and over 1,500 citations.
Manubot's pull request-based contribution model balances the goals of making the project open to everyone and maintaining scientific accuracy.
All contributions are reviewed, discussed, and formally approved on GitHub before text updates appear in the public-facing manuscript at <https://greenelab.github.io/covid19-review/>.
Continuous integration (CI) seamlessly combines author-produced text and figures with automatically generated and updated statistics and figures from external data sources and the manuscript's own content.
In addition, the authors who initially launched this project included Manubot developers who had prior successes using Manubot for massively open and traditional manuscript.

We could use `analyze-ms-stats/manuscript_stats.json` from #1019 to automate the exact number of references.  The last sentence is meant to acknowledge that we also selected Manubot in part because of familiarity and our existing bias toward it (driven by prior successes).

Author: I need to get that PR finalized -- will work on that next!

Author: Once #1034 merges, we should be able to make all of these dynamic. There are several places throughout the text where they come up.

This approach emphasizes examining the methods used, assignment (whether the study was observational or randomized), assessment, results, interpretation, and how well the study extrapolates [@doi:10.5014/ajot.60.4.367].
Contributors were also invited to contribute or edit text using GitHub's pull request system.
These contributions were not strictly defined and could range from minor corrections to punctuation and grammar to large-scale additions of text.
Finally, a small number of contributors (the authors of this paper) contributed technical expertise, either through the development of standardized approaches to the evaluation of papers based on the MAARIE Framework [@isbn:9780781774260], the writing of code to generate manuscript figures, or the addition of features to Manubot.
All of these additions were also submitted as pull requests, either to the COVID-19 review repository or to an external repository, as appropriate.

Each pull request was reviewed and approved by at least one other contributor before being merged into the main branch.
We tagged potential reviewers based on the introductions they had contributed in order to encourage participation.

> Authorship was determined based on CRediT [@url:https://casrai.org/credit].

Reviewer: Reviewer 3 suggested using footnotes for URLs, and I recommend that for URL citations as well.  The URL citations do not translate well to the conference LaTeX template and are quite difficult to fix manually.  I think we could keep URL citations for the small number of URLs that refer to an article with a specific author, title, and date (like `url:https://blogs.sciencemag.org/pipeline/archives/2021/07/19/too-many-papers`) and convert the rest.

Pandoc does support footnotes (see https://github.com/manubot/rootstock/issues/352).  We would need to see how they show up in the long HTML document and the converted tex file.

```suggestion
Authorship was determined based on the Contributor Roles Taxonomy [@url:https://casrai.org/credit].
```

Reviewer: We could follow up in that rootstock issue to discuss modifying 02.delete-me.md.  There is always a gray area about how much to demonstrate in that example document.  We try to not fully reproduce the pandoc documentation when the feature is pure pandoc but also need to demonstrate common use cases.

For example, beyond building just a PDF, Manubot also renders the manuscript in HTML, DOCX, and now, LaTeX (in a more limited capacity).
The interactive HTML manuscript format offers several advantages over a static PDF to harmonize available resources that we were able to apply to specific problems of COVID-19.
The integration of scite into the HTML build makes references more manageable by visually representing whether their results are contested or whether they have been corrected or retracted.
Cross-referencing different pieces of the manuscript, such as cited preprints with reviews stored in an appendix, is another interactive option presented by HTML.
The DOCX format was preferred by most non-technical users for reviewing the final version of the manuscript and was useful for creating submissions to biological journals.
Additionally, because of the heavy emphasis on Word processing in biology, Manubot's ability to generate DOCX outputs was expanded to allow users to generate DOCX files containing only a section of the manuscript.

> In our case, where the full project is nearly 100,000 words, this allows individual pieces to be shared more easily.

Author: ```suggestion
In our case, where the full project is nearly 150,000 words, this allows individual pieces to be shared more easily.
```

In turn, by seeking to adapt Manubot to allow for broader participation in open publishing from fields where computational training in tools like version control is uncommon, we made a number of improvements that are expected to increase its appeal to researchers from all backgrounds.
Manubot provided a way for contributors from a variety of backgrounds, including early-career researchers, to join a massive collaborative project while demonstrating their individual contributions to the larger work and gaining experience with version control.
The licensing and infrastructure also provide an opportunity for individuals to adapt from this project to create their own snapshots of the COVID-19 literature that derive from, but are not wholly identical to, the primary versions of these reviews.
This project suggests that massive online open publishing efforts can indeed advance scholarship through inclusion [@doi:10.5334/kula.63], including during the extreme challenges presented by the COVID-19 pandemic.

> Some limitations, however, arose in inclusion, either due to the unique circumstances of the pandemic or as the result of challenges broadly associated with open publishing projects.

Author: ```suggestion
Some challenges, however, arose in efforts to include an academically diverse set of authors, either due to the unique circumstances of the pandemic or as the result of challenges broadly associated with open publishing projects.
```

The barriers to entry posed by git and GitHub likely still reduced participation from individuals who might have otherwise been interested.

> Using pull requests as a tool for writing text is unfamiliar to many or most scientists, and the review process can be slow, which might cause interested contributors to lose interest.

Author: ```suggestion
Using pull requests as a tool for writing text is also unfamiliar to many or most scientists, and the review process can be slow, which might cause interested contributors to lose interest.
```

Given that our approach hinged on these version control tools, it is likely that our group of contributors was biased towards those who were interested in or experienced with computational tools.
The trajectory of the pandemic itself also likely influenced participation: engagement waned over the course of the pandemic as labs opened back up and researchers were able to return to their work, and we recruited very few medical doctors to the project, which is unsurprising given the load on medical professionals during this time.
Engagement that waxes and wanes is, in however, typical when writing massively open online papers [@doi:10.5334/kula.63].
Adding features such as spell-check has improved user friendliness, and additional features such as automatically checking the formatting of citations could further improve the usability of this tool.
In the future, a formal study of participation could allow for quantification of these biases and improved efforts to foster inclusion.

Some other limitations are challenges associated with of massively open online papers in general.

> With such a large amount of text, it is not possible to keep all sections of the manuscript up to date.

Author: ```suggestion
With such a large amount of text, it is not possible to keep all sections of the manuscript up to date at all times.
```

Readers are not able to distinguish when each section was updated, and even a functionality such as GitHub's blame feature would not distinguish minor changes from substantive updates to the text.
Additionally, while much of the data and statistics update automatically, the text itself required updating by human experts.

> This asynchrony could potentially introduce incompatibility between the figures and the surrounding text.

Author: ```suggestion
This asynchronicity could potentially introduce incompatibility between the figures and the surrounding text.
```

As a result, the lead authors of each paper often spent several weeks prior to submission updating the text to reflect new developments in each area.
In the future, it may be possible to streamline this process through integration with a tool such as CoronaCentral [@doi:10.1073/pnas.2100766118] to automatically identify relevant, high-impact papers that need to be included, although expertise would still be required to incorporate them.
Another challenge involves tracking preprints as they are reviewed or critiqued, revised, and potentially published.
While updating the content of the manuscript would likely fall to human contributors, automatic detection of published versions of preprints could be integrated in the future (as developed by [@doi:10.1101/2021.03.04.433874]).
These challenges are exacerbated by the scale of the infodemic, but developing solutions would benefit future projects with more typical trends in literature.
Similarly, outputting machine readable summaries of key information in the manuscripts developed by the Consortium could reduce our contribution to the infodemic.

> As it stands, the integration of Compact Uniform Resource Identifier does make a step in this direction, as it could be used to extract relationships among clinical trials, genes, publications, and other entities.

Author: ```suggestion
As it stands, the integration of Compact Uniform Resource Identifier does make a step in this direction, as it could be used to extract relationships among clinical trials, genes, publications, and other entities.
Thus, the experience of using Manubot for a massive non-technical project has laid the foundation for future additions to enhance user experience and inclusivity.
```

As many other efforts have described, the publishing rate of formal manuscripts and preprints about COVID-19 has been unprecedented [@doi:10.1053/j.ackd.2020.08.003], and efforts to review the body of COVID-19 literature are faced with an ever-expanding corpus to evaluate.
In the case of the seven manuscripts produced by the COVID-19 Review Consortium, Manubot will allow for continuous updating of the manuscripts as the pandemic enters its second year and the landscape shifts with the emergence of promising therapeutics and vaccines [@individual-pharmaceuticals; @individual-vaccines].
These manuscripts pull data from external sources, allowing for information and visualizations to be updated daily using CI.
This computational approach allows for some of the updating process to be off-loaded so that domain experts can focus on the broader implications of new information as it emerges.
As a result, centralizing, summarizing, and critiquing data and literature broadly relevant to COVID-19 can help to expedite the interdisciplinary scientific process that is currently happening at an advanced pace.

> As of September 13, 2021, almost 3,000 commits have been made to the manuscript across 575 merged pull requests.

Reviewer: Woah!  Where did the 3,000 commits come from?  Is this one of the stats you now track in a JSON file?  It looks like it distinguishes between commits associated with pull requests versus all 5,000 commits to master (which includes rootstock and non-content commits).

Reviewer: Never mind, now I see `"manuscript_commits": 2886` in `covid19-review-stats.json`

The interactive HTML manuscript format offers several advantages over a static PDF to harmonize available resources that we were able to apply to specific problems of COVID-19.
The integration of scite into the HTML build makes references more manageable by visually representing whether their results are contested or whether they have been corrected or retracted.
Cross-referencing different pieces of the manuscript, such as cited preprints with reviews stored in an appendix, is another interactive option presented by HTML.
Additionally, because of the heavy emphasis on Word processing in biology, Manubot's ability to generate DOCX outputs was expanded to allow users to generate DOCX files containing only a section of the manuscript.
In our case, where the full project is nearly 100,000 words, this allows individual pieces to be shared more easily.
Finally, the preliminary addition of LaTeX output is useful for researchers from computational fields who submit papers in TeX format and removes the step of reformatting markdown prior to submission. 

| *Additions to Manubot* | 
|:-----------------|

> | - Figures and statistics that automatically update based on external data sources |
> | - Check spelling and compile a list of spelling errors, along with their locations in the document(s) | 
> | - Citation of clinical trials as {% raw %}`[@clinicaltrials:NCT04707534]`{% endraw %} (required also adding Zotero support for identifiers) |  
> | - Smart citations through integration with scite |
> | - Export into LaTeX format | 

Author: ```suggestion
| - CI scripts to regularly download external data sources, generate figures and statistics, and store them in a GitHub branch that is read when Manubot builds the latest manuscript |
| - Zotero extension to report more relevant clinical trial metadata from https://clinicaltrials.gov |
| - Manubot feature to support citing any Compact Uniform Resource Identifier, such as clinicaltrials or ncbigene
| - Integration with scite to track retractions, corrections, and notices of concern |
| - CI spell-checking reports that post as pull request comments |
| - Improved support for Pandoc's LaTeX output |
| - Building individual sections of a larger manuscript as standalone documents alongside the complete manuscript |
```

Reviewer 2: Could rephrase to:

> Zotero translator to extract citation metadata from ClinicalTrials.gov.

Want to stress _citation_ metadata, since reader may not be familiar with Zotero.

Otherwise suggestion looks great. An impressive list of contributions!

> Table: List of extensions added to Manubot's functionalities for the COVID-19 Collaborative Review.

Author: > * otero extension to report more relevant clinical trial metadata from https://clinicaltrials.gov
> 
> * Manubot feature to support citing any Compact Uniform Resource Identifier, such as `clinicaltrials` or `ncbigene`
> 
> * Integration with scite to track retractions, corrections, and notices of concern
> 
> * CI spell-checking reports that post as pull request comments
> 
> * Improved support for Pandoc's LaTeX output
> 
> * Building individual sections of a larger manuscript as standalone documents alongside the complete manuscript

The current project was managed through GitHub in the repository greenelab/covid19-review using Manubot [@doi:10.1371/journal.pcbi.1007128] to continuously generate a version of the manuscript online in HTML^[https://greenelab.github.io/covid19-review] and PDF^[https://greenelab.github.io/covid19-review/manuscript.pdf] formats.
The Manubot framework facilitated a massive collaborative review on an urgent topic.
This project demonstrates that Manubot can be applied to a project where not all contributors had expertise or even experience working with version control pipelines.
This effort has produced not only seven literature reviews on topics relevant to the COVID-19 pandemic, but has also generated cyberinfrastructure for training novice users to interact with GitHub and extended functionalities of Manubot to provide more of the benefits of What You See Is What You Get platforms such as Google Docs (@tbl:manubot-addons).
Open publishing thus allowed us to harness the domain expertise of a large group of non-technical users to respond to the flood of COVID-19 publications.

Several existing and new features in Manubot aid in responding to the challenges posed by the infodemic.

> While Manubot manuscripts are written in markdown, they can be rendered in several formats that provide different advantages for a variety of useres.

Author: ```suggestion
While Manubot manuscripts are written in markdown, they can be rendered in several formats that provide different advantages for a variety of users.
```

Engagement that waxes and wanes is, in however, typical when writing massively open online papers [@doi:10.5334/kula.63].
Adding features such as spell-check has improved user friendliness, and additional features such as automatically checking the formatting of citations could further improve the usability of this tool.
In the future, a formal study of participation could allow for quantification of these biases and improved efforts to foster inclusion.

Some other limitations are challenges associated with of massively open online papers in general.
With such a large amount of text, it is not possible to keep all sections of the manuscript up to date.
Readers are not able to distinguish when each section was updated, and even a functionality such as GitHub's blame feature would not distinguish minor changes from substantive updates to the text.
Additionally, while much of the data and statistics update automatically, the text itself required updating by human experts.

> This ansynchrony could potentially introduce incompatibility between the figures and the surrounding text.

Author: ```suggestion
This asynchrony could potentially introduce incompatibility between the figures and the surrounding text.
```

> Similarly, in line with the colloration-related challenges of the project, some authors returned to update their text, while others did not.

Author: ```suggestion
Similarly, in line with the collaboration-related challenges of the project, some authors returned to update their text, while others did not.
```

For example, beyond building just a PDF, Manubot also renders the manuscript in HTML, DOCX, and now, LaTeX (in a more limited capacity).
The interactive HTML manuscript format offers several advantages over a static PDF to harmonize available resources that we were able to apply to specific problems of COVID-19.
The integration of scite into the HTML build makes references more manageable by visually representing whether their results are contested or whether they have been corrected or retracted.
Cross-referencing different pieces of the manuscript, such as cited preprints with reviews stored in an appendix, is another interactive option presented by HTML.
Additionally, because of the heavy emphasis on Word processing in biology, Manubot's ability to generate DOCX outputs was expanded to allow users to generate DOCX files containing only a section of the manuscript.
In our case, where the full project is nearly 100,000 words, this allows individual pieces to be shared more easily.
Finally, the preliminary addition of LaTeX output is useful for researchers from computational fields who submit papers in TeX format and removes the step of reformatting markdown prior to submission. 

> <!-- To Do: Add table listing additions to manubot-->

Reviewer: Here is an initial summary:

- CI scripts to regularly download external data sources, generate figures and statistics, and store them in a GitHub branch that is read when Manubot builds the latest manuscript
- Zotero extension to report more relevant clinical trial metadata from https://clinicaltrials.gov
- Manubot feature to support citing any Compact Uniform Resource Identifier, such as `clinicaltrials` or `ncbigene`
- Integration with scite to track retractions, corrections, and notices of concern
- CI spell-checking reports that post as pull request comments
- Improved support for Pandoc's LaTeX output
- Building individual sections of a larger manuscript as standalone documents alongside the complete manuscript

This feels redundant with the existing text.  Do we agree with the reviewer that the summary is helpful?

Author: Yours is better than mine! I do like the idea of a table, because I have to keep looking through the text to jog my memory about the full list!

Manubot provided a way for all contributors, including early-career researchers, to join a massive collaborative project, demonstrating their individual contributions to the larger work and gaining experience with version control.
This project shows that massive online open publishing efforts can indeed advance scholarship through inclusion [@doi:10.5334/kula.63], including during the extreme challenges presented by the COVID-19 pandemic.
However, the barriers to entry posed by git and GitHub likely still reduced participation from individuals who might have otherwise been interested.
Additionally, using an approach that hinged on these tools likely biased our contributors towards those who were interested in or experienced with computational tools.
The trajectory of the pandemic itself also likely influenced participation: engagement waned over the course of the pandemic as labs opened back up and researchers were able to return to their work, and we recruited very few medical doctors to the project, which is unsurprising given the load on medical professionals during this time.

> These are limitations that we hope to improve through future work on Manubot.

Author: I think I need to move line 282 to the paragraph below to make this more clear, we cannot fix the pandemic with Manubot!

> These are limitations that we hope to improve through future work on Manubot.

Author: Just wanted to remark that I'm in the process of reorganizing these paragraphs and adding ideas from the discussion here, and I'll push the changes later this morning!

For example, while we automated the updating of data and statistics, the text itself required updating by human experts.
Given that engagement waxed and waned in this project, as is typical for these types of undertakings [@doi:10.5334/kula.63], this presented challenges at times.
Some authors returned to update their text, while others did not.
As a result, the lead authors of each paper often spent several weeks prior to submission updating the text to reflect new developments in each area.
In the future, it may be possible to streamline this process through integration with a tool such as CoronaCentral [@doi:10.1073/pnas.2100766118] to automatically identify relevant, high-impact papers that need to be included, although expertise would still be required to incorporate them.
Such a functionality would serve to address another major limitation to this project: ultimately, the scale of the infodemic made it incredibly challenging to stay on top of the COVID-19 literature, even with engagement from a large number of collaborators from diverse backgrounds.

> 

Reviewer: Let's consider additional limitations of the Manubot approach.  Ryan brought up a good one at https://github.com/greenelab/covid19-review/pull/947#pullrequestreview-649162037  Readers of the HTML manuscript cannot tell what parts of the paper are up to date and which are stale.  A savvy reader could use GitHub's blame feature to determine when each sentence or figure was last edited, but that information is not displayed in the rendered manuscript and doesn't distinguish between typo corrections and substantive updates.

A related idea is that we have automating figures and statistics that are summarized by static contributed text.  There is a risk that the data in the figure or statistics can change and the related text will be wrong or inconsistent.  We sometimes tried to work around this by writing "As of date XYZ" in the text, but the core limitation remains.

The pull request model can sometimes be tedious.  Adding even a single sentence requires making a pull request, waiting for an approval, and waiting for the CI build to finish.  That has deterred some colleagues from joining Manubot projects in the past.

There is no good way to provide general comments on a manuscript.  Comments can be added to pull requests, but only to the modified content.  Hypothesis comments can be added anywhere but are not easy to integrate back into the manuscript like pull request comments or suggestions.

Any other limitations we've faced?

Author: Another challenge is definitely that we haven't pivoted completely away from DOCX. I have been manually adding commits that people send me either in emails or in docx attachments (using --author). 

Also, there are silly issues like Casey's setting automatically remove EOL whitespace, so every time he opens a PR, it has a huge impact in terms of line numbers. We can fix this by automating this for every PR, but I haven't gotten to it!

Another one-- we don't have a good way to link up preprints with their published papers. I sometimes do this randomly while checking the text -- I'll open a preprint to confirm that I'm not changing the meaning with some edit, and then find that it has been published. I think we can also add this thanks to work David Nicholson has been doing.

Reviewer: Other limitations from the reviews we could mention:
- We integrated with the Mount Sinai preprint reviews but any other relevant efforts (e.g. preLights).  There are other potential contributors who are/were reviewing COVID-19 preprints.  How could we build upon their existing efforts?
- scite slows down HTML load times (https://github.com/greenelab/covid19-review/issues/1015).  Could we load the badges only on request?
- Formal user study of Manubot versus other collaborative writing approaches, provide evidence as to how usable (or not) Manubot is for non-technical users.  Bias among current authors toward technical users (related to the point above about DOCX).
- Machine readable/actionable summaries of key information in the review manuscripts as a way to combat the infodemic (as opposed to creating yet another set of manuscripts to read)

Author: I believe these are all addressed in the new draft text!

{%- endif %}

### Acknowledgements {.unnumbered}

> We are grateful to Josh Nicholson and Milo Mordaunt for their support with the scite plugin, to Milton Pividori for feedback on code, and to David Nicholson for the suggestion and feedback to enable the reporting of the locations of spelling errors in the spell-checker tool.

Author: ```suggestion
We are grateful to Josh Nicholson and Milo Mordaunt for their support with the scite plugin, to Milton Pividori for feedback on code, to David Nicholson for the suggestion and feedback to enable the reporting of the locations of spelling errors in the spell-checker tool, and to Consortium members Alex Lee and Christian Brueffer for feedback on the code and text, respectively.
```

We thank Nick DeVito for assistance with the Evidence-Based Medicine Data Lab COVID-19 TrialsTracker data.

> This research was supported by the Gordon and Betty Moore Foundation award GBMF 4552 (HMR, DSH, CSG), the National Institutes of Health award R01HG010067 (CSG), and the John W. and Jeanne M. Rowe Center for Research in Virology (AG).

Author: ```suggestion
This research was supported by the Gordon and Betty Moore Foundation award GBMF 4552 (HMR, DSH, CSG), the National Institutes of Health award R01HG010067 (HMR, CSG), and the John W. and Jeanne M. Rowe Center for Research in Virology (AG).
```
